pymongo/periodic_executor.py

```diff
@@ -91,7 +93,15 @@ def open(self) -> None:
             thread.daemon = True
             self._thread = weakref.proxy(thread)
             _register_executor(self)
-            thread.start()
+            # Mitigation to RuntimeError firing when thread starts on shutdown
+            # https://github.com/python/cpython/issues/114570
+            try:
+                thread.start()
+            except RuntimeError as e:
+                if str(e) == _THREAD_START_ON_SHUTDOWN_ERR:
```

Comment:
What do you think of `if "interpreter shutdown" in str(e) or sys.is_finalizing():`? This could be more future proof.

Reply:
Wasn't sure if you meant explicitly "interpreter shutdown", so I've left it with the current, longer string.

Reply:
Yes I meant "interpreter shutdown" since it's a more broad but still safe.